cfgov/v1/documents.py

```diff
@@ -193,15 +201,19 @@ def filter(
         self.filter_categories(categories=categories)
         self.filter_language(language=language)
         self.filter_date(from_date=from_date, to_date=to_date)
+        self._count = None
 
     def search(self, title=""):
         """Perform a search for the given title"""
         self.search_title(title=title)
-        return self.search_obj[0 : self.count()].to_queryset(keep_order=True)
+        self._count = None
+        return self.search_obj
 
     def count(self):
         """Return the search object's current result count"""
-        return self.search_obj.count()
+        if self._count is None:
+            self._count = self.search_obj.count()
+        return self._count
```

Comment:
All the changes in this file are to cache the ES results count where possible. Currently we make multiple calls to retrieve the count but we don't need to do that unless the filtering criteria changes.